### DISCUSSION TITLE
How to get the build id, for example: https://project-buildID.now.sh/

### DISCUSSION THREAD

**Discussion by harm-meijer** [vercel/vercel#3880] [General] [answered]
I am trying to run a next.js app on now and pre hydrate api data.

In pages/api/[type].js I have an api that proxies requests from client to another server. So in client code I have something like: 

    const url = toUrl(`${API_URL}/products`, {
      query
    })

When I run it locally then API_URL is `http://localhost/api` so when the code runs on the server or on the client it will both be localhost but when run on now I need it to be `https://project-buildID.now.sh/api`, the problem is that I don't know what the build id will be and can't find it in `process.env`, dumped them in the api with `res.end(JSON.stringify(process.env, undefined, 2))` but the build id is not in there.

**Comment by coetry** [comment, marked as answer]
Have you considered using a production or branch domain for this? That will give you a consistent URL that's predictable that can stay static and it will automatically be assigned to the latest build. 

https://vercel.co/docs/v2/custom-domains#assigning-a-domain-to-a-git-branch

Then in your `next.config.js` you can configure the `API_URL` value.

```
// next.config.js
module.exports = {
  env: {
    API_URL: process.env.NODE_ENV === "production" 
      ? "https://production-domain/api" 
      : "http://localhost:3000/api"
  }
}
```

**Reply by harm-meijer** [reply]
Thank you for your reply, there will be lots of builds I guess and `https://production-domain/api` should point to the url of the current build.

I can have it point to a fixed domain but if I change the api code then cypress tests will fail because it is not using the current code.

**Reply by coetry** [reply]
@harm-meijer correct, it will always point to the latest build automatically!

**Reply by harm-meijer** [reply]
But what if I need it to point to the current build (like the build for the pull request)?

**Reply by coetry** [reply]
You can retrieve that information from the `host` header, in `getInitialProps`, but that will opt-out of the static optimization for the page and require a lambda to be invoked every time a user visits the page. So its definitely a tradeoff you'll need to consider. 

https://vercel.co/docs/v2/network/headers#host

**Reply by harm-meijer** [reply]
Thank you. I'll try to wrap that code in an if statement that checks if the code is running from production url or not. The optimization is already gone because I have getInitialProps in _app.js that dispatches an action to set the query in the redux store.
